pkg/client/block/client.go

```diff
@@ -106,7 +106,7 @@ type blockReplayClient struct {
 	onStartQueryClient client.BlockQueryClient
 
 	// cometClient is the CometBFT client used to get ABCI info for chain version.
-	cometClient cometclient.Client
+	cometClient *cometclienthttp.HTTP
```

Comment:
We should keep the dependency of `cometclient.Client` which is the interface that `cometclienthttp.HTTP` implements. This helps us to mock it properly